src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.skia.cs

```diff
@@ -38,24 +59,496 @@ private void UpdateTextBoxView()
+			if (point.Properties.HasPressedButton && !point.Properties.IsRightButtonPressed)
+			{
+				var index = displayBlock.Inlines.GetIndexForTextBlock(point.Position - new Point(displayBlock.Padding.Left, displayBlock.Padding.Top));
+				var selectionInternalStart = _selectionEndsAtTheStart ? _selection.start + _selection.length : _selection.start;
+				SelectInternal(selectionInternalStart, index - selectionInternalStart);
+			}
+		}
+	}
+
+	protected override void OnRightTapped(RightTappedRoutedEventArgs e)
+	{
+		base.OnRightTapped(e);
+		e.Handled = true;
+
+		if (!FeatureConfiguration.TextBox.UseOverlayOnSkia)
+		{
+			if (_contextMenu is null)
+			{
+				_contextMenu = new MenuFlyout();
+				_contextMenu.Opened += (_, _) => UpdateDisplaySelection();
+
+				_flyoutItems.Add("Cut", new MenuFlyoutItem { Text = "Cut", Command = new TextBoxCommand(CutSelectionToClipboard), Icon = new SymbolIcon(Symbol.Cut) });
+				_flyoutItems.Add("Copy", new MenuFlyoutItem { Text = "Copy", Command = new TextBoxCommand(CopySelectionToClipboard), Icon = new SymbolIcon(Symbol.Copy) });
+				_flyoutItems.Add("Paste", new MenuFlyoutItem { Text = "Paste", Command = new TextBoxCommand(PasteFromClipboard), Icon = new SymbolIcon(Symbol.Paste) });
+				// undo/redo
+				_flyoutItems.Add("Select All", new MenuFlyoutItem { Text = "Select All", Command = new TextBoxCommand(SelectAll), Icon = new SymbolIcon(Symbol.SelectAll) });
+			}
+
+			_contextMenu.Items.Clear();
+
+			if (_selection.length == 0)
+			{
+				_contextMenu.Items.Add(_flyoutItems["Paste"]);
+				// undo/redo
+				_contextMenu.Items.Add(_flyoutItems["Select All"]);
+			}
+			else
+			{
+				_contextMenu.Items.Add(_flyoutItems["Cut"]);
+				_contextMenu.Items.Add(_flyoutItems["Copy"]);
+				_contextMenu.Items.Add(_flyoutItems["Paste"]);
+				// undo/redo
+				_contextMenu.Items.Add(_flyoutItems["Select All"]);
+			}
+
+			_contextMenu.ShowAt(this, e.GetPosition(this));
+		}
+	}
+
+	protected override void OnDoubleTapped(DoubleTappedRoutedEventArgs e)
+	{
+		base.OnDoubleTapped(e);
+		e.Handled = true;
+
+		_wasJustDoubleTapped = true;
+	}
+
+	partial void OnPointerPressedNative(PointerRoutedEventArgs e)
+	{
+		if (!FeatureConfiguration.TextBox.UseOverlayOnSkia && !_wasJustDoubleTapped && (!e.GetCurrentPoint(null).Properties.IsRightButtonPressed || SelectionLength == 0))
+		{
+			var displayBlock = TextBoxView.DisplayBlock;
+			var index = displayBlock.Inlines.GetIndexForTextBlock(e.GetCurrentPoint(displayBlock).Position - new Point(displayBlock.Padding.Left, displayBlock.Padding.Top));
+			Select(index, 0);
+		}
+	}
+
+	private (int start, int length) FindChunkAt(int index, bool right)
+	{
+		if (Text.GetHashCode() != _cachedChunks.hashCode)
+		{
+			GenerateChunks();
+		}
+
+		var i = 0;
+		foreach (var chunk in _cachedChunks.chunks)
+		{
+			if (chunk.start < index && chunk.start + chunk.length > index || chunk.start == index && right || chunk.start + chunk.length == index && !right)
```

Comment:
Let's adjust formatting, but also use parenthesis to avoid relying on the precedence of operators.
```suggestion
			if (chunk.start < index 
			   && chunk.start + chunk.length > index 
			   || chunk.start == index 
			   && right 
			   || chunk.start + chunk.length == index 
			   && !right)
```